### DISCUSSION TITLE
Building a controller example modification

### DISCUSSION THREAD

**Discussion by kosmylo** [e2nIEE/pandapower#1275] [Q&A]
I have noticed in the "building a controller" example that the p_mw of the storage comes from a dataframe, which is given beforehand. What I would like to do is to have a storage controller, which is related to other values in my network, such as PV power. For example to give a charge command when the PV power exceeds the load demand. Is this possible to do something like that?

**Comment by rbolgaryn** [comment]
Hi @kosmylo ,

yes, absolutely.

You can define your own storage controller that reacts to changes of other elements.
You can take a look at the characteristic controller, hopefully, it can help by providing a reference for this kind of behavior.
Also, the USetTapControl has a similar concept.

https://github.com/e2nIEE/pandapower/blob/develop/pandapower/control/controller/characteristic_control.py
https://github.com/e2nIEE/pandapower/blob/develop/pandapower/control/controller/trafo/USetTapControl.py
https://github.com/e2nIEE/pandapower/blob/develop/tutorials/u_set_tap.ipynb

Best regards,
Roman

**Comment by Aivlyz** [comment]
Did you manage to do so? I want to try to do the same thing.
